crates/ln-dlc-node/src/tests/mod.rs

```diff
@@ -1,6 +1,368 @@
+        let data_dir = random_tmp_dir().join(name);
+
+        let seed = Bip39Seed::new().expect("A valid bip39 seed");
+
+        let mut ephemeral_randomness = [0; 32];
+        thread_rng().fill_bytes(&mut ephemeral_randomness);
+
+        let address = {
+            let listener = TcpListener::bind("127.0.0.1:0").unwrap();
+            listener.local_addr().expect("To get a free local address")
+        };
+
+        let node = Node::new(
+            name.to_string(),
+            Network::Regtest,
+            data_dir.as_path(),
+            address,
+            ELECTRS_ORIGIN.to_string(),
+            seed,
+            ephemeral_randomness,
+            user_config,
+        )
+        .await;
+
+        let bg_processor = node.start().await?;
+        mem::forget(bg_processor); // to keep it running
+
+        tracing::debug!(%name, info = ?node.info, "Node started");
+
+        Ok(node)
+    }
+
+    async fn fund(&self, amount: bitcoin::Amount) -> Result<()> {
+        let starting_balance = self.get_confirmed_balance()?;
+        let expected_balance = starting_balance + amount.to_sat();
+
+        let address = self
+            .wallet
+            .get_new_address()
+            .map_err(|_| anyhow!("Failed to get new address"))?;
+
+        fund_and_mine(address, amount).await;
+
+        while self.get_confirmed_balance()? < expected_balance {
+            let interval = Duration::from_millis(200);
+
+            self.sync();
+
+            tokio::time::sleep(interval).await;
+            tracing::debug!(
+                ?interval,
+                "Checking if wallet has been funded after interval"
+            )
+        }
+
+        Ok(())
+    }
+
+    fn get_confirmed_balance(&self) -> Result<u64> {
+        let balance = self.wallet.inner().get_balance()?;
+
+        Ok(balance.confirmed)
+    }
+
+    /// Initiates the opening of a channel _and_ waits for the channel
+    /// to be usable.
+    async fn open_channel(
+        &self,
+        peer: NodeInfo,
+        amount_us: u64,
+        amount_them: u64,
+    ) -> Result<ChannelDetails> {
+        let temp_channel_id =
+            self.initiate_open_channel(peer, amount_us + amount_them, amount_them)?;
+
+        let channel_details = tokio::time::timeout(Duration::from_secs(10), async {
+            loop {
+                if let Some(details) = self
+                    .channel_manager
+                    .list_usable_channels()
+                    .iter()
+                    .find(|c| c.counterparty.node_id == peer.pubkey)
+                {
+                    break details.clone();
+                }
+
+                tracing::debug!(
+                    %peer,
+                    temp_channel_id = %hex::encode(temp_channel_id),
+                    "Waiting for channel to be usable"
+                );
+                tokio::time::sleep(Duration::from_millis(500)).await;
```

Comment:
I'm not sure I understand why this work. Is a channel not only usable once we have X confirmation? And this function doe snot create blocks, so why does it work?

Reply:
I believe that all our configs support 0-conf channels at the time of writing this code:

```rust
UserConfig {
    // ...
    // we want to accept 0-conf channels from the coordinator
    manually_accept_inbound_channels: true,
    // ...
}
```

I think that should not be the case in general, so a nice thing to do here might be to mine as many blocks as the channel config says and then start looping.

Reply:
I've updated the documentation and added a TODO.